controller/controller_server/sync_lock.py

```diff
@@ -0,0 +1,56 @@
+import threading
+from collections import defaultdict
+
+from controller.common.csi_logger import get_stdout_logger
+from controller.controller_server.errors import ObjectAlreadyProcessingError
+
+logger = get_stdout_logger()
+
+ids_in_use = defaultdict(set)
+ids_in_use_lock = threading.Lock()
+
+
+def _add_to_ids_in_use(lock_key, object_id):
+    ids_in_use[lock_key].add(object_id)
+
+
+def _remove_from_ids_in_use(lock_key, object_id):
+    ids_in_use[lock_key].remove(object_id)
+
+
+def _add_object_lock(lock_key, object_id, action_name):

```

Comment:
when I suggested these functions to be private, I didn't mean they have to be outside the class.
in fact, I think it would be better to place them inside the class, to reduce parameters

Reply:
but if you think it is better to keep them here, that's also fine

Reply:
I moved them back